nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentFeature.java

```diff
@@ -35,5 +35,6 @@
 @Deploy("org.nuxeo.ecm.platform.comment.api")
 @Deploy("org.nuxeo.ecm.platform.query.api")
 @Deploy("org.nuxeo.ecm.platform.comment")
+@Deploy("org.nuxeo.ecm.platform.comment.tests:OSGI-INF/notification-contrib.xml")
 public class CommentFeature implements RunnerFeature {
 }
```

Comment:
perhaps i missed some thing but why add the `notification-contribution` on `CommentFeature` the feature responsible of notification is `NotificationCommentFeature`: 

i tried to rollback this part (i mean removr the `notification-contrib` from commentFeature and put it in the `NotificationCommentFeature` i ran the whole tests, there some tests that failed (mainly the `AbstractTestAnnotationService` stack)  because `if (!hasComments(session, document, commentAuthor.getName())` is null to fix it:

- i create  the users 
```
        CommentUtils.createUser("jdoe");
        CommentUtils.createUser("bob");
```

even if i did one test still failed on the `AbstractTestAnnotationService` it is because there is no comment auhtor and you code try to find it

```
if (StringUtils.isBlank(author)) {
            String[] contributors = (String[]) docModel.getPropertyValue("dc:contributors");
            author = contributors[0];
        }
        return Framework.getService(UserManager.class).getPrincipal(author);
```
but as the comment is created using `doPriviliege...` the `dc:contributors` will be system and the  Framework.getService(UserManager.class).getPrincipal(author); will returns `null` i don't why but i think its better te put the deploy notification contribution and fix the remained tests:

`org.nuxeo.ecm.restapi.server.jaxrs.comment.AbstractAnnotationAdapterTest#testUpdateAnnotation`
`org.nuxeo.ecm.restapi.server.jaxrs.comment.AbstractAnnotationAdapterTest#testUpdateExternalAnnotation`
`org.nuxeo.ecm.restapi.server.jaxrs.comment.AbstractCommentAdapterTest#testUpdateExternalComment`
`org.nuxeo.ecm.platform.comment.impl.TestTreeCommentManager#shouldDeleteComment`

i think it is the same bug there is no `author` just call the `CommentUtils.createUser()`

from web-ui i think we will not have an issue because they will send the author

but with REST API any other client, we will have an issue if the caller don't put the author and `Framework.getService(UserManager.class).getPrincipal(author);` returns `null` in the case `system`

Reply:
or if we don't want to create the needed users -> at least i think it will better to rollback the modification of moving the contribution and create a specific contib to enable/disable the autosubscirber my idea is to keep separated the Notification and Comment Feature, Notification Comment Feature should know about Comment Feature but not the  opposite.

Reply:
@troger any feedback on my two comments see above :)

Reply:
Context extras:
Disabling notification on all tests previous to this ticket was done to preserve testing behavior (ignore autosubscription and avoid creating users everywhere).
Disabling notifications was necessary as in the new behavior, notifications are enabled by default.
Then to test auto-subscription, notifications are re-enabled  by test.

Reply:
yes i see already the whole test of notification comment/annotation as the subscription on the test and create the needed user. having the auto-subscription from my point of view dont break the existing behaviour if the user is already subscribed the new code will not change anything and my idea is to keep the use of the 2 feature simple as if you import the `NotificationCommentFeature` outer the comment code the user if it should think about adding the contribution to enable the auto-subscribe or he should do it himself which means he should see the source code to understand how he should do the things but if we enable it on `NotificationCommentFeature` there is not need to investigate to how the feature works. this is my point of view :)